### DISCUSSION TITLE
APIs being triggered before data provider loads

### DISCUSSION THREAD

**Discussion by mathewhealthi** [Budibase/budibase#9069] [Help] [answered]
We have a page in our app where a data provider (which pulls data from postgres) has a repeater nested in it and each repeated section triggers two APIs based on a certain unique parameter in the each repeated section. 

Since updating our app to the latest version, we have noticed that as soon as the page loads, the 2 APIs are triggered without any parameters sent through them which causes multiple error messages to pop up. 

In the older version of the app, the API would trigger only once the data provider & repeater loaded, and worked seamlessly. Is this a bug or is there a fix for this?

**Comment by melohagan** [comment]
Hey @mathewhealthi 

Which version worked, and which version did the issue present itself? Then I can dig a bit further.

**Comment by mathewhealthi** [comment, marked as answer]
Hey @melohagan , the version 2.1.39 works as expected, while the problem occurs on version 2.2.9. 

Answer: Upgrade to version **2.2.19**

**Reply by melohagan** [reply]
Thanks @mathewhealthi We believe that it could be related to the recently introduced skeleton loader. We'll have a look into it, but there may not be a fix soon because it's quite a complex part of the codebase.

**Reply by mathewhealthi** [reply]
Hi @melohagan, is there any workaround for this issue? We're struggling to figure out how to avoid error messages cropping up each time

**Reply by melohagan** [reply]
Hey @mathewhealthi there was a release yesterday that may have fixed these issues if you want to upgrade to the latest version: https://github.com/Budibase/budibase/releases

Should be **2.2.15**

**Reply by mathewhealthi** [reply]
Oh that's good to hear @melohagan! When would these newer versions become available on the Budibase Cloud-hosted version of Budibase? We can currently only update apps up to 2.2.9.

**Reply by melohagan** [reply]
Hey @mathewhealthi 

That's been deployed now - do you want to try updating and let me know if it resolves your original issue?

**Reply by mathewhealthi** [reply]
Hi @melohagan , the issue has been resolved now, it works as expected. Thanks so much for the help!

**Reply by melohagan** [reply]
Great to hear it!
